e2etest/l2tests/l2.go

```diff
@@ -469,6 +472,23 @@ var _ = ginkgo.Describe("L2", func() {
 				framework.ExpectError(err, fmt.Sprintf("metallb_speaker_announced present in node: %s", p.Spec.NodeName))
 			}
 
+			ginkgo.By("validating the speaker doesn't publish layer2 metrics after deleting the service")
+			err = cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err)
+
+			gomega.Eventually(func() error {
+				speakerMetrics, err := metrics.ForPod(controllerPod, advSpeaker, metallb.Namespace)
+				if err != nil {
+					return err
+				}
+
+				err = metrics.ValidateGaugeValue(1, "metallb_speaker_announced", map[string]string{"node": advSpeaker.Spec.NodeName, "protocol": "layer2", "service": fmt.Sprintf("%s/%s", f.Namespace.Name, svc.Name)}, speakerMetrics)
+				if err == nil {
+					return fmt.Errorf("metallb_speaker_announced present in node: %s", advSpeaker.Spec.NodeName)
+				}
+
+				return nil
+			}, 1*time.Minute, 1*time.Second).Should(gomega.BeNil())
```

Comment:
do we really need eventually to confirm we don't get metric after the svc is deleted ?